py_zipkin/zipkin.py

```diff
@@ -709,3 +713,125 @@ def create_http_headers_for_new_span(context_stack=None, tracer=None):
         'X-B3-Flags': '0',
         'X-B3-Sampled': '1' if zipkin_attrs.is_sampled else '0',
     }
+
+
+def is_valid_trace_id(candidate):
+    return candidate and re.match(r'(?:[0-9a-f]{16}|[0-9a-f]{32})$', candidate)
+
+
+def is_valid_span_id(candidate):
+    return candidate and re.match(r'[0-9a-f]{16}$', candidate)
+
+
+def _extract_zipkin_attrs_from_single_header(b3_header, sample_rate,
+                                             use_128bit_trace_id):
+    # b3={TraceId}-{SpanId}-{SamplingState}-{ParentSpanId}
+    # where the last two fields are optional.
+    bits = b3_header.split('-')
+    if len(bits) == 1:
+        # See if it's a valid encoding of (only) sample decision
+        if bits[0] == 'false' or bits[0] == '0':
```

Comment:
I don't think `false` is a valid sample header. Reading https://github.com/openzipkin/b3-propagation/blob/master/RATIONALE.md it seems like it could be either `0`, `1` or `d` but the description of this case is very unclear.
cc @adriancole

Reply:
Makes sense that single `b3` header would not have the legacy baggage of the old `X-B3-Sample` header.

Removing that leniency.